tests/integration/tests/configure_charm.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Pre-test configuration of a prometheus-juju-exporter testing model."""
+
+import logging
+from base64 import b64encode
+from pathlib import Path
+
+import yaml
+from zaza import model
+
+from .test_charm import wait_for_model_settle
+
+logger = logging.getLogger(__name__)
+
+JUJU_LOCAL_DATA = Path("/home/ubuntu/.local/share/juju/")
+ACCOUNT_FILE = JUJU_LOCAL_DATA.joinpath("accounts.yaml")
+CONTROLLER_FILE = JUJU_LOCAL_DATA.joinpath("controllers.yaml")
+
+
+def setup_juju_credentials() -> None:
+    """Configure prometheus-juju-exporter with required juju credentials.
+
+    Credentials are pulled form config files of a nested controller deployed by
+    'juju-local' charm during the tests.
+    """
+    controller_units = model.get_units("juju-local")
+    if len(controller_units) < 1:
```

Comment:
This function does lots of things we can split into several functions.
get_unit
get_controller_credentials
get_controller_endpoint_details

Reply:
“Don’t Use a Comment When You Can Use a Function or a Variable”
― Robert C. Martin, [Clean Code: A Handbook of Agile Software Craftsmanship]

Reply:
I'm going to jump to this discussion. I really like clean code patter such as "splitting to multiple functions", because functions like that are easier to test (multiple simple functions then one big), reusable and more readable. Is there any more reason why it's good to have multiple functions?

IMO this function is part of testing, so it is not tested itself and it will not be reused anywhere. So there is only one thing and that is the readability of the function, which I think is preserved. I think that several functions here are not mandatory from the point of view of functionality (testing, reusing, etc.), so the only thing left is readability, and that is more a subjective view.

Reply:
As a general rule functions should do one thing. It is obvious that this function does more than one thing because there are three parts with comments. Instead of comments I would prefer readable function names. In this case it is not a deal breaker. There is nothing wrong on the architectural standpoint.